[charts/datadog] Allow granular read secret roles (#640)
DataDog/helm-charts: charts/datadog/CHANGELOG.md

# Datadog changelog

## 2.33.8

> * Added new value `datadog.secretBackend.enableGlobalPermissions` which when set to `false` will not allow Datadog agents to read all secrets in all clusters - defaults to `true` (current behaviour)
> * Added new value `datadog.secretBackend.roles` which will create `Role` and `RoleBinding` for each namespace defined - allowing for opt-in read permissions for secrets in those namespaces

Reviewer: ```suggestion
* Add the `datadog.secretBackend.enableGlobalPermissions` value, which when set to `false`, does not allow Datadog agents to read all secrets in all clusters. By default, this defaults to `true`.
* Add the `datadog.secretBackend.roles` value, which creates `Role` and `RoleBinding` for each namespace defined. Allows opt-in read permissions for secrets in those namespaces.
```

Author: Done
